content/movie-review-pt1.md

```diff
@@ -0,0 +1,450 @@
+
+Traits describe an abstract interface that types can implement. If a trait defines a function `bark()` and a type then adopts that trait, the type must then implement the `bark()` function.
+
+[Attributes](https://doc.rust-lang.org/rust-by-example/attribute.html) add metadata to a type and can be used for many different purposes.
+
+When you add the [`derive` attribute](https://doc.rust-lang.org/rust-by-example/trait/derive.html) to a type and provide one or more supported traits, code is generated under the hood to automatically implement the traits for that type. We'll provide a concrete example of this shortly.
+
+## Representing instructions as a Rust data type
+
+Now that we've covered the Rust basics, let's apply them to Solana programs.
+
+More often than not, programs will have more than one function. For example, you may have a program that acts as the backend for notes. Assume this program accepts instructions for creating a new note, updating an existing note, and deleting an existing note. 
+
+Since instructions have discrete types, they're usually a great fit for an enum data type.
+
+```rust
+enum NoteInstruction {
+    CreateNote {
+        title: String,
+        body: String,
+        id: u64
+    },
+    UpdateNote {
+        title: String,
+        body: String,
+        id: u64
+    },
+    DeleteNote {
+        id: u64
+    }
+}
+```
+
+Notice that each variant of the `NoteInstruction` enum comes with embedded data that will be used by the program to accomplish the taks of creating, updating, and deleting a note, respectively.
+
+## Deserialize instruction data
+
+Instruction data is passed to the program as a byte array, so you need a way to deterministically convert that array into an instance of the instruction enum type.
+
+In previous modules, we used Borsh for client-side serialization and deserialization. To use Borsh program-side, we use the `borsh` crate. This crate provides traits for `BorshDeserialize` and `BorshSerialize` that you can apply to your types using the `derive` attribute. 
+
+To make deserializing instruction data simple, you can create a struct representing the data and use the `derive` attribute to apply the `BorshDeserialize` trait to the struct. This implements the methods defined in `BorshDeserialize`, including the the `try_from_slice` method that we'll be using to deserialize the instruction data.
+
+Remember, the struct itself needs to match the structure of the data in the byte array.
+
+```rust
+#[derive(BorshDeserialize)]
+struct NoteInstructionPayload {
+    id: u64,
+    title: String,
+    body: String
+}
+```
+
+Once this struct has been created, you can create an implementation for your instruction enum that defines and implements a function that accepts the instruction data as an argument and returns the appropriate instance of the enum with the deserialized data.
```

Comment:
@jamesrp13 I don't love the repetition of "... an implementation ... implements ..."

Mainly tho, this sentence feels a bit run-on - valid, but given the conceptual depth & importance, maybe a small breather, e.g.
```
Once this struct has been created, you can create an implementation of your instruction enum. This implementation defines and implements...
```

Reply:
Updated. lmk if you're okay with the update